### ISSUE TITLE
Memory leaks

### ISSUE THREAD

**Issue by Maxatma** [WXGBridgeQ/SwiftPullToRefresh#17]
Class used RefreshView:
```
import SwiftPullToRefresh
import MaterialComponents


final class KZFooter: RefreshView {
    private var loader: MDCActivityIndicator!
    
    override public init(style: Style, height: CGFloat, action: @escaping () -> Void) {
        loader             = MDCActivityIndicator(frame: CGRect(x: 0, y: 0, width: 32, height: 32))
        loader.cycleColors = [UIColor.dcmBrightLightBlue]
        loader.radius      = 15
        
        super.init(style: style, height: height, action: action)
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    override func didUpdateState(_ isRefreshing: Bool) {

        if isRefreshing {
            addSubview(loader)
            loader.center = CGPoint(x: frame.width / 2, y: frame.height / 2)
            loader.startAnimating()
        } else {
            loader.removeFromSuperview()
            loader.stopAnimating()
        }
    }
    
    override func didUpdateProgress(_ progress: CGFloat) {

    }
}
```
Where I use this class:

```
final class CurrentVC : UIViewController {
    var viewModel: CurrentVM!
    var footer: KZFooter!
    var header: KZFooter!
    @IBOutlet weak var table: UITableView!


    override func viewDidLoad() {
        super.viewDidLoad()

        footer = KZFooter(style: .autoFooter, height: 57) { [weak self] in
            guard let sself = self else { return }
            sself.viewModel!.loadNextPage.next(())
        }
        
        table.spr_setCustomFooter(footer)
        
        header = KZFooter(style: .header, height: 57) { [weak self] in
            guard let sself = self else { return }
            sself.viewModel!.refresh.next(())
        }

        table.spr_setCustomHeader(header)
	}
}
```

It don't deallocate and every time I reopen screen it create more instances of KZFooter

<img width="914" alt="screen shot 2018-02-19 at 12 36 38 pm" src="https://user-images.githubusercontent.com/7135226/36374244-554c9cae-1573-11e8-8eb9-7f3136ef5c2c.png">
<img width="1387" alt="screen shot 2018-02-19 at 12 49 02 pm" src="https://user-images.githubusercontent.com/7135226/36374245-5583c206-1573-11e8-8ec1-884adb6ec9dc.png">

<img width="228" alt="screen shot 2018-02-19 at 12 36 34 pm" src="https://user-images.githubusercontent.com/7135226/36374243-55160662-1573-11e8-8f71-0859ad15d13b.png">

Any ideas how to fix that

**Comment by WXGBridgeQ** [comment]
@Maxatma 
Sorry for the late reply, thanks for finding the bug, I fixed it in the latest version (2.3.0), you can have a check again.

**Comment by WXGBridgeQ** [comment]
Please try to use 2.3.1
